build/Targets/Imports.targets

```diff
@@ -492,6 +492,14 @@
     </PropertyGroup>
   </Target>
 
+  <Target Name="BeforeBuild" Condition="'$(OS)' == 'Windows_NT'">
+    <ValidateBuildEnvironment
```

Comment:
They represent different checks:

- Imports.targets: This guards every build of our code base. It provides a polite failure when the wrong version of MSBuild is used.
- Powershell: Here we are discovering MSBuild instances on the machine and need to filter to 15.1 only. This covers items like running Build.cmd, CI machines, etc ... It helps us not hit the error that is issued in Imports.targets.